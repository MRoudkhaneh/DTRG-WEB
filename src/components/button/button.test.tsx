import { Button } from '.'
import { render, screen } from '@testing-library/react'

describe('Button component', () => {
  it('Should render children', () => {
    render(<Button>Click me</Button>)
    const child = screen.getByText(/Click me/i)
    expect(child).toBeVisible()
  })
  it('Should be disabled', () => {
    render(<Button disabled>Click me</Button>)
    const child = screen.getByRole('button')
    expect(child.className).toContain('opacity-50')
  })
  it('Should be loading', () => {
    render(<Button loading>Click me</Button>)
    const child = screen.getByRole('button')
    expect(child.className).toContain('cursor-wait')
  })
})
