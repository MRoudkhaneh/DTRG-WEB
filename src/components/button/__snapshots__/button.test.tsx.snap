// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button SHOUD Match <Button/> TO SNAPSHOT 1`] = `
<button
  className=" px-2  rounded focus:outline-none shadow flex items-center justify-center  undefined hover:opacity-80"
  id="button"
/>
`;
